spec/lib/migrations/load_rate_reference_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+require 'rake'
+require 'roo'
+
+RSpec.describe 'Load Rating Regions Task', :type => :task do
+  
+  context "rate_reference:update_rating_regions" do
+    before :all do
+      Rake.application.rake_require "tasks/migrations/load_rate_reference"
+      Rake::Task.define_task(:environment)
+    end
+    it 'expect not raise errors' do
+      expect { invoke_task }.not_to raise_error
+    end
+    
+    it 'expect to read xlsx file' do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
```

Comment:
I mentioned on a commit that I added that this can be separated into a single action, but I took it a step further and removed the Roo usage entirely.

If we consider what the test is doing and what the service is doing as two separate actions. Then there's no reason for the test to have any knowledge of what the service ACTUALLY does - it only cares about the results. This lets our test focus on what the actual desired outcomes are, while our service can focus on the actual how.

-- The exception is some types of unit testing, in which case you may be deliberately mocking external dependencies - requiring your test to know a great deal about the implementation.